Day_2_Software_engineering_best_practices/05_documentation_git.html

```diff
@@ -0,0 +1,196 @@
+]
+
+---
+
+# What `git` is used for?
+
+.center[![](../img/git/writing.png)]
+
+---
+
+count: false
+# What `git` is used for?
+
+.center[![](../img/git/coding.png)]
+
+---
+count: false
+# What `git` is used for?
+
+* Keep track of history
+* Safe development of new features
+* Swap between versions
+* **Enable collaborative coding, writing, ...**
+
+---
+
+# Concepts of `git`
+
+### Common commands
+
+* `$ git init`: initialize a `git` repository.
+* `$ git add file`: start tracking changes for a file.
+* `$ git rm file`: stop tracking changes for a file.
+* `$ git status`: get information regarding the status of the repository.
+* `$ git commit`: get a snapshot of the changes of the files tracked.
+* `$ git diff`: see the difference between different commit for the files.
+
+---
+
+# Concepts of `git`
+
+### Branches
+
+* `$ git branch name_branch`: create a new branch.
+
+* `$ git checkout name_branch`: switch to a specific branch.
+
+* `$ git merge another_branch`: bring the changes of `another_branch` into the current branch.
+
+---
+
+# Concepts of `git`
+
+### Remote
+
+* `$ git remote add origin git://some_remote.git`: add a remote named `origin` pointing somewhere.
+
+* `$ git push origin name_branch`: push the local changes of `name_branch` into the remote `origin`.
+
+* `$ git pull origin name_branch`: pull the changes from the remote `origin` and the branch `name_branch`.
+
+---
+
+# Concepts of `git`
+
+### Summary
+
+.center[![:scale 50%](../img/git/git-transport.png)]
+
+---
+
+# Install and configure `git`
+
+### Install
+
+You can refer to the following install
+[instructions](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+
+### Configure
+
+Before starting using `git`, you will have configure it:
+
+```bash
+$ git config --global user.name "John Doe"
+$ git config --global user.email johndoe@example.com
+```
+
+---
+
+# Live example
+
+### 1. Initialize
+
+Go to `spectra_analysis` directory and initialize the `git` repository. It will create an hidden `.git` folder.
+
+```bash
+$ git init
+```
+
+### 2. Add barely minimum information
+
+* `README.md`: contains information regarding the package.
+* `.gitignore`: contains which file types to automatically skip. Examples can be found [there](https://github.com/github/gitignore). For us, we can select [`Python.gitignore`](https://github.com/github/gitignore/blob/master/Python.gitignore).
+
+---
+
+# Live example
+
+### 3. Track files
+
+We can start to track files.
+
+```bash
+$ git add .gitignore README.md
+```
+
+We can also add directly a folder.
+
+```bash
+$ git add .
+```
+
+### 4. Check the status
+
+You can use `status` to get information about the repository.
+
+```bash
+$ git status
+```
+
+---
+
+# Live example
+
+### 5. Commit changes
+
+We can take a snapshot of state of the repository by committing.
+
+```bash
+$ git commit -m 'Initial commit'
+```
+
+### 6. Check the commit history
+
+We can now check the history.
+
+```bash
+$ git log
+```
+
```

Comment:
do we still want to do the next step here for adding the python files and commiting them?